backend/server.js

```diff
@@ -1,28 +1,93 @@
 import express from 'express'
 import bodyParser from 'body-parser'
-import cors from 'cors'
+import cors from 'cors' 
 import mongoose from 'mongoose'
+import crypto from 'crypto'
+import bcrypt from 'bcrypt-nodejs'
 
 const mongoUrl = process.env.MONGO_URL || "mongodb://localhost/authAPI"
-mongoose.connect(mongoUrl, { useNewUrlParser: true, useUnifiedTopology: true })
+mongoose.connect(mongoUrl, { useNewUrlParser: true, useUnifiedTopology: true, useCreateIndex: true })
 mongoose.Promise = Promise
 
-// Defines the port the app will run on. Defaults to 8080, but can be 
-// overridden when starting the server. For example:
-//
-//   PORT=9000 npm start
+const User = mongoose.model('User', {
+  name: {
+    type: String,
+    unique: true
+  },
+  email: {
+    type: String,
+    unique: true
+  },
+  password: {
+    type: String,
+    required: true
+  },
+  accessToken: {
+    type: String,
+    default: () => crypto.randomBytes(128).toString('hex')
+  }
+})
+
 const port = process.env.PORT || 8080
 const app = express()
 
-// Add middlewares to enable cors and json body parsing
 app.use(cors())
 app.use(bodyParser.json())
 
-// Start defining your routes here
-app.get('/', (req, res) => {
-  res.send('Hello world')
+const authenticateUser = async (req , res, next) => {
+  try {
+    const user = await User.findOne({
+      accessToken: req.header('Authorization'),
+    })
+    if (user) {
+      req.user = user
+      next()
+    } else {
+      res
+        .status(401)
+        .json({ loggedOut: true, message: 'Not loged in, please try again' })
+    }
+  } catch (err) {
+    res
+      .status(403)
+      .json({ message: 'Access token is wrong or missing', errors: err })
+  }
+}
+
+app.get('/', async (req, res) => {
+  const users = await User.find()
+  res.json(users)
 })
```

Comment:
Hmm, is there a reason you want the user to be able to see all other users (with hashed passwords and accessTokens and all)? 🤔